src/client/Client.cc

```diff
@@ -1670,6 +1689,48 @@ void Client::dump_mds_requests(Formatter *f)
   }
 }
 
+void Client::get_caps_asok_interface(std::string relpath, int mask, Formatter *f)
+{
+  filepath path(relpath);
+  InodeRef inref;
+  UserPerm perms {0,0};
+  ldout(cct, 20) << __func__  << " Requesting caps " << ccap_string(mask)
+	         << " on the file " << relpath << dendl;
+  int r = path_walk(path, &inref, perms, false);
+  if (r < 0) {
+    lderr(cct) << __func__ << " path resolution failed. path = " << relpath
+	       << " errno = " << -r << " (" << strerror(-r) << ")" << dendl;
+    f->dump_stream("path resolution failed") << -r << " (" << strerror(-r) << ")";
+    return;
+  }
+
+  Inode *in = inref.get();
+  for (auto &p : in->caps) {
+    mds_rank_t mds = p.first;
+    Cap &cap = p.second;
+
+    auto session = mds_sessions.at(mds);
+
+    int revoking = cap.implemented & ~cap.issued;
+    int msg_flags = 0;
+    int flushing = 0;
+    ceph_tid_t flush_tid = 0;
+    msg_flags |= MClientCaps::FLAG_SYNC;
+    int cap_used = get_caps_used(in) | in->caps_dirty();
+    int wanted = mask;
+    int retain = (cap.issued | cap.implemented);
+    ldout(cct, 20) << __func__  << " caps issued " << ccap_string(cap.issued)
+	           << " cap implemented " << ccap_string(cap.implemented)
+	           << " cap revoking " << ccap_string(revoking)
+	           << " cap used " << ccap_string(cap_used)
+	           << " cap wanted " << ccap_string(mask)
+                   << " on the file " << relpath << dendl;
+    send_cap(in, session.get(), &cap, msg_flags, cap_used, wanted, retain,
```

Comment:
Why always request this from MDS ? Could we just check whether the client has already been issued the caps wanted, then we can do nothing. 

Or could we just call `get_caps()` directly here ?